syntax = "proto3";
package pkg;

service WhisperModel {
  rpc SpeechToText(SpeechToTextRequest) returns (SpeechToTextResponse) {}
}

message SpeechToTextRequest {
  string task = 1;
  bytes audio_file = 2;
  string initial_prompt = 3;
  string language_target = 4;
  string output_format = 5;
  bool word_timestamps = 6;
}

message SpeechToTextResponse {
  string text = 1;
}